src/include/common/concurrent_pointer_vector.h

```diff
@@ -0,0 +1,293 @@
+#pragma once
```

Comment:
If I'm not mistaken, this means ConcurrentPointerVector and its associated test are not used; could those be removed for a smaller diff? Thanks!

Reply:
Yes thats correct. I shall remove them